recommendations.md

```diff
@@ -50,14 +50,23 @@ MUST avoid forwarding these user credentials accidentally (see
 
 ### Authorization Code Grant {#ac}
 
-Clients MUST prevent injection (replay) of authorization codes into the
-authorization response by attackers. Public clients MUST use PKCE [@!RFC7636] to
-this end. For confidential clients, the use of PKCE [@!RFC7636] is RECOMMENDED.
-With additional precautions, described in (#nonce_as_injection_protection),
-confidential clients MAY use the OpenID Connect `nonce` parameter and the
-respective Claim in the ID Token [@!OpenID] instead. In any case, the PKCE
-challenge or OpenID Connect `nonce` MUST be transaction-specific and securely
-bound to the client and the user agent in which the transaction was started.
+Clients MUST prevent misuse of authorization codes and authorization code
+injection attacks (see (#code_injection)) using one of the following options:
+
+ * Public clients MUST use PKCE [@!RFC7636] to this end, as motivated in
+   (#pkce_as_injection_protection). 
+ * For confidential clients, the use of PKCE [@!RFC7636] is RECOMMENDED, as it
+   provides a strong protection against misuse and injection of authorization
+   codes as described in (#pkce_as_injection_protection) and, as a side-effect,
+   prevents CSRF even in presence of strong attackers as described in
+   (#csrf_countermeasures). 
+ * With additional precautions, described in (#nonce_as_injection_protection),
```

Comment:
Shouldn't we mention `c_hash` here as well? Since checking the nonce by itself doesn't prevent authorization code injection for `response_type=code+id_token`

Reply:
It does when you take the ID Token from the backend into consideration as well (which is always a good idea). This is described in (#csrf_countermeasures).

Reply:
that seems worth pointing out explicitly then

Reply:
There's no need for c_hash actually. The backend-ID-Token will contain the wrong nonce. (#csrf_countermeasures) recommends to check both.